torch/utils/data/_utils/collate.py

```diff
@@ -130,56 +249,4 @@ def default_collate(batch):
             >>> default_collate([[0, 1], [2, 3]])
             [tensor([0, 2]), tensor([1, 3])]
     """
```

Comment:
nit and unsure if we want this, but we can also add an example of extending `default_collate` for when users do not want to extend via `Type`. Maybe add an example like:
```python
def custom_collate(batch):
    if batch is ...:  # Some custom condition
        ...
    else:  # Fall back to `default_collate`
        return default_collate(batch)
```

Feel free to ignore this if you don't think this will be useful.

Reply:
I agree. Such an example would be nice for users.

Reply:
Make sense. I am adding two options to extend `default_collate`.